rest/authorization-server/src/main/java/org/eclipse/sw360/rest/authserver/security/Sw360AuthorizationServerConfiguration.java

```diff
@@ -70,13 +71,13 @@ public void configure(AuthorizationServerSecurityConfigurer oauthServer) throws
     @Override
     public void configure(ClientDetailsServiceConfigurer clients) throws Exception {
         clients.inMemory()
-                .withClient(clientId)
+                .withClient(CONFIG_CLIENT_ID == null ? clientId : CONFIG_CLIENT_ID)
```

Comment:
I am not sure whether this is the correct point to inject the overriding configuration. It does not modify the field in the class and might lead to confusing behaviour if the field is used somewhere else. Maybe you can overwrite it on the `@Value` annotation level?

(Same for the secret)

Reply:
I tried to overwrite the @value annotation directly.
But this is not possible.

As result I created a new @PostConstruct method `postSw360AuthorizationServerConfiguration` to overwrite the @value annotation fields.